### PR TITLE
[SILGen] eliminate SILGen for var parameters

### PR DESCRIPTION
**Pull request by dduan**
#### What's in this pull request?

var parameter is no longer a thing in Swift 3. So we can remove the SILGen for it.
#### Resolved bug number: ([SR-](https://bugs.swift.org/browse/SR-))

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

File: lib/SILGen/SILGenProlog.cpp [hunk 1]
```diff
@@ -284,20 +284,6 @@ struct ArgumentInitHelper {
         gen.B.createDebugValueAddr(loc, argrv.getValue(), {vd->isLet(), ArgNo});
       else
         gen.B.createDebugValue(loc, argrv.getValue(), {vd->isLet(), ArgNo});
-    } else {
-      // If the variable is mutable, we need to copy or move the argument
```
Comment (slavapestov) [inline lib/SILGen/SILGenProlog.cpp]: Can you add an assert that vd->isLet() instead of just falling through the if/else if?

**Comment by dduan** [comment]
Done.

**Comment by slavapestov** [comment]
@swift-ci Please test and merge

**Comment by slavapestov** [comment]
Thanks!

**Comment by slavapestov** [comment]
@swift-ci Please test

**Comment by dduan** [comment]
Failing tests are failing for every pull request. Safe to assume it's unrelated.

**Pull request merged**